compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -218,6 +219,15 @@ impl<'a> ModCollector<'a> {
                 }
             }
 
+            let is_low_level_function =
+                function.attributes().function.as_ref().map_or(false, |func| func.is_low_level());
+            if !in_stdlib && is_low_level_function {
+                let error = DefCollectorErrorKind::LowLevelFunctionOutsideOfStdlib {
```

Comment:
It feels like this check would be better in name resolution rather than collection. `extract_meta` is most likely the best place for this where we can check against a `FunctionKind`. Also, checking only here we are only looking at free functions. This will still pass with this change:
```
fn main(x: Field) -> pub u1 {
    let dummy = Dummy { x };
    let bits = dummy.to_le_bits(x, 100);
    bits[0]
}

struct Dummy {
    x: Field
}

impl Dummy {
    #[builtin(to_le_bits)]
    fn to_le_bits(self, _x: Field, _bit_size: u32) -> [u1] {}
}
```

Reply:
Nice, yeah this is a much better place to put it. Thanks!